controllers/limitador_controller.go

```diff
@@ -99,6 +93,16 @@ func (r *LimitadorReconciler) Reconcile(eventCtx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	// Reconcile Status
+	if err := r.reconcileStatus(ctx, limitadorObj); err != nil {
+		switch err.Error() {
+		case "resource not ready":
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
As a future enhancement, instead of requeue, the controller should watch for owned (referenceOwner) deployments and when the deployment says it is available, a new reconciliation loop will be triggered. If for some reason, the lmitador becomes unavailable (because it crashes or whatever), this controller will never know and the status will be available until something changes the spec of the liimitador CR

Reply:
Requeueing should happen also when the deployment exists but reports not enough replicas 
```
limitadorInstance.Status.ReadyReplicas < 1
```